metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesProvider.scala

```diff
@@ -75,32 +87,63 @@ final class TestSuitesProvider(
 
   override def onChange(docs: TextDocuments, file: AbsolutePath): Unit =
     docs.documents.headOption.foreach {
-      case doc if isEnabled => updateTestCases((file, doc))
+      case doc if isExplorerEnabled => updateTestCases((file, doc))
```

Comment:
I see that you've changed `isEnabled` to `isExplorerEnabled`, won't this result in stale test cases exposed as lenses?

Reply:
AFAICS `updateTestCases` is just for test explorer use since it only invokes a client command?  Thanks for the comment though, it helped me spot [2c7e711](https://github.com/scalameta/metals/pull/4569/commits/2c7e711b205dc2d19a7b3f5bf0d3a56b5d6593fc)

Reply:
I think code lenses will be refreshed every time compilation finishes so they will get correctly updated.